### PR TITLE
[clang][OpenMP] Implement `isOpenMPExecutableDirective`

### PR DESCRIPTION
**Pull request by kparzysz**
What is considered "executable" in clang differs slightly from the OpenMP's "executable" category. In addition to the executable category, subsidiary directives, and OMPD_error are considered executable.

Implement a function that performs that check.

### CODE REVIEW

File: clang/include/clang/Basic/OpenMPKinds.h [hunk 1]
```diff
@@ -368,6 +368,14 @@ bool needsTaskBasedThreadLimit(OpenMPDirectiveKind DKind);
 /// is restricted only to memory order clauses of "OMPC_acquire",
 /// "OMPC_relaxed" and "OMPC_seq_cst".
 bool checkFailClauseParameter(OpenMPClauseKind FailClauseParameter);
+
+/// Checks if the specified directive is considered as "executable". This
+/// combines the OpenMP categories of "executable" and "subsidiary", plus
+/// any other directives that are should be treated as executable.
```
Comment (alexey-bataev) [inline clang/include/clang/Basic/OpenMPKinds.h:374]: that ~~are~~ should

File: clang/include/clang/Basic/OpenMPKinds.h [hunk 2]
```diff
@@ -368,6 +368,14 @@ bool needsTaskBasedThreadLimit(OpenMPDirectiveKind DKind);
...
+/// \param DKind Specified directive.
+/// \return true - if the above condition is met for this directive
```
Comment (alexey-bataev) [inline clang/include/clang/Basic/OpenMPKinds.h:376]: condition ~~is~~ met

**Review by alexey-bataev — APPROVED** [review]
LG with nits

File: clang/include/clang/Basic/OpenMPKinds.h (hunk 2 above)
Comment (kparzysz) [reply clang/include/clang/Basic/OpenMPKinds.h:376]: Are you sure?  "...condition is met for this directive..." sounds correct to me.

File: clang/include/clang/Basic/OpenMPKinds.h (hunk 1 above)
Comment (kparzysz) [reply clang/include/clang/Basic/OpenMPKinds.h]: Done

File: clang/include/clang/Basic/OpenMPKinds.h (hunk 2 above)
Comment (alexey-bataev) [reply clang/include/clang/Basic/OpenMPKinds.h:376]: Up to you

**Pull request merged**